tests/beta/test_default_tensor.py

```diff
@@ -1754,3 +1754,39 @@ def test_hermitian(self, theta, phi, varphi, monkeypatch, rep, tol):
             )
         ) / 16
         assert np.allclose(var, expected, atol=tol, rtol=0)
+
+
+class TestWiresIntegration:
+    """Test that the device integrates with PennyLane's wire management."""
+
+    def make_circuit_expval(self, wires, representation):
+        """Factory for a qnode returning expvals using arbitrary wire labels."""
+        dev = qml.device("default.tensor", wires=wires, representation=representation)
+        n_wires = len(wires)
+
+        @qml.qnode(dev)
+        def circuit():
+            # use modulo to make circuit independent of number of wires
+            qml.RX(0.5, wires=wires[0 % n_wires])
+            qml.Hadamard(wires=wires[3 % n_wires])
+            qml.RY(2., wires=wires[1 % n_wires])
+            if n_wires > 1:
+                qml.CNOT(wires=[wires[0 % n_wires], wires[1 % n_wires]])
+            return [qml.expval(qml.PauliZ(wires=w)) for w in wires]
+
+        return circuit
+
+    @pytest.mark.parametrize("representation", ["exact", "mps"])
+    @pytest.mark.parametrize("wires1, wires2", [(['a', 'c', 'd'], [2, 3, 0]),
+                                                ([-1, -2, -3], ['q1', 'ancilla', 2]),
+                                                (['a', 'c'], [3, 0]),
+                                                ([-1, -2], ['ancilla', 2]),
+                                                (['a'], ['nothing']),
+                                                ])
```

Comment:
No worries if this is a massive overhead now, but would it also make sense to have the "vanilla" case as well, e.g., `[0, 1, 2]`?

Reply:
I think that's a one-line change :grinning:

Reply:
Added!